x-pack/plugins/security_solution/server/lib/detection_engine/rule_response_actions/endpoint_response_action.ts

```diff
@@ -5,81 +5,155 @@
+import type {
+  RuleResponseEndpointAction,
+  ProcessesParams,
+} from '../../../../common/api/detection_engine';
+import type { KillOrSuspendProcessRequestBody } from '../../../../common/endpoint/types';
+import { getErrorProcessAlerts, getIsolateAlerts, getProcessAlerts } from './utils';
+import type { AlertsAction, ResponseActionAlerts } from './types';
 import type { EndpointAppContextService } from '../../../endpoint/endpoint_app_context_services';
-import { getProcessAlerts, getIsolateAlerts, getErrorProcessAlerts } from './utils';
-
-import type { ResponseActionAlerts, AlertsAction } from './types';
 
-export const endpointResponseAction = (
+export const endpointResponseAction = async (
   responseAction: RuleResponseEndpointAction,
   endpointAppContextService: EndpointAppContextService,
-  { alerts }: ResponseActionAlerts,
-  experimentalFeatures: ExperimentalFeatures
-) => {
+  { alerts }: ResponseActionAlerts
+): Promise<void> => {
+  const logger = endpointAppContextService.createLogger(
+    'ruleExecution',
+    'automatedResponseActions'
+  );
+  const ruleId = alerts[0][ALERT_RULE_UUID];
+  const ruleName = alerts[0][ALERT_RULE_NAME];
+  const logMsgPrefix = `Rule [${ruleName}][${ruleId}]:`;
   const { comment, command } = responseAction.params;
+  const agentIds = uniq(map(alerts, 'agent.id'));
+  const errors: string[] = [];
+  const responseActionsClient = endpointAppContextService.getInternalResponseActionsClient({
+    agentType: 'endpoint',
+    username: 'unknown',
+  });
+
+  const automatedProcessActionsEnabled =
+    endpointAppContextService.experimentalFeatures.automatedProcessActionsEnabled;
+
+  const processResponseActionClientError = (err: Error, endpointIds: string[]): Promise<void> => {
+    errors.push(
+      `attempt to [${command}] host IDs [${endpointIds.join(', ')}] returned error: ${err.message}`
+    );
 
-  const commonData = {
-    comment,
-    command,
-    rule_id: alerts[0][ALERT_RULE_UUID],
-    rule_name: alerts[0][ALERT_RULE_NAME],
-    agent_type: 'endpoint' as const,
+    return Promise.resolve();
   };
 
-  if (isIsolateAction(responseAction.params)) {
-    const alertsPerAgent = getIsolateAlerts(alerts);
-    each(alertsPerAgent, (actionPayload) => {
-      return endpointAppContextService.getActionCreateService().createActionFromAlert(
-        {
-          ...actionPayload,
-          ...commonData,
-        },
-        actionPayload.endpoint_ids
+  const response: Array<Promise<void>> = [];
+
+  switch (command) {
+    case 'isolate':
+      response.push(
+        Promise.all(
+          Object.values(getIsolateAlerts(alerts)).map(
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            ({ endpoint_ids, alert_ids, parameters, error, hosts }: AlertsAction) => {
+              logger.info(
+                `${logMsgPrefix} [${command}] [${endpoint_ids.length}] agent(s): ${stringify(
```

Comment:
I think that this may be the issue with failing test, `stringify` doesn't exist - I assume it should be `JSON.stringify`  ?

Reply:
Looks like I was having some IDE issues. I have corrected this. `stringify` import is missing